File: SysAsistenciaAn/src/main/java/pe/edu/upeu/asistencia/models/Comentario.java
package pe.edu.upeu.asistencia.models;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalTime;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Table(name = "comentarios")
public class Comentario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "comentario_id")
    private Long id;

    @Column(name = "usuario")
    private Long usuarioId;

    @Column(name = "tipo_entidad")
    private String tipoEntidad;

    @Column(name = "contenido")
    private String contenido;

    @JsonFormat(pattern = "yyyy-MM-dd")
    @Basic(optional = false)
    @Column(name = "fecha", nullable = false)
    @Temporal(TemporalType.DATE)
    private LocalDate fecha;

    // Getters and setters
}

File: AsistenciaUPeUJCR/app/src/main/java/pe/edu/upeu/asistenciaupeujcr/modelo/Comentario.kt
package pe.edu.upeu.asistenciaupeujcr.modelo

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "comentarios")
data class Comentario(
    @PrimaryKey(autoGenerate = true)
    var comentarioId: Long = 0,
    var usuario: Long,
    var tipoEntidad: String,
    var contenido: String,
    var fecha: String
)
